models/Asset.php

```diff
@@ -893,39 +893,34 @@ public function getSiblings(): array
                 }
                 $list->setOrderKey('filename');
                 $list->setOrder('asc');
-                $this->siblings = $list->getAssets();
+                $this->siblings = $list;
             } else {
-                $this->siblings = [];
+                $list = new Asset\Listing();
+                $list->setAssets([]);
+                $this->siblings = $list;
             }
         }
 
         return $this->siblings;
     }
 
-    public function hasSiblings(): ?bool
+    public function hasSiblings(): bool
     {
         if (is_bool($this->hasSiblings)) {
-            if (($this->hasSiblings && empty($this->siblings)) || (!$this->hasSiblings && !empty($this->siblings))) {
-                return $this->getDao()->hasSiblings();
-            } else {
-                return $this->hasSiblings;
-            }
+            return $this->hasSiblings;
         }
 
-        return $this->getDao()->hasSiblings();
+        return $this->hasSiblings = $this->getDao()->hasSiblings();
```

Comment:
Somehow the suggestions don't for me at the moment, when I select the `if` too, but this can be replaced with (and the `if` can be removed then): 

```suggestion
        return $this->hasSiblings ??= $this->getDao()->hasSiblings();
```